File: js/5525.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const M = +inputs[1];
const S = inputs[2];

let i = 0;
let cnt = 0;
let answer = 0;

while (i <= M - 2) {
  if (S.substring(i, i + 3) === 'IOI') {
    i += 2;
    cnt += 1;
    if (cnt === N) {
      answer += 1;
      cnt -= 1;
    }
  } else {
    i += 1;
    cnt = 0;
  }
}

console.log(answer);

File: c++/5525.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);
  int N, M;
  string S;
  cin >> N >> M >> S;

  int i = 0;
  int cnt = 0;
  int answer = 0;

  while (i <= M - 2) {
    if (S.substr(i, 3).compare("IOI") == 0) {
      i += 2;
      cnt += 1;
      if (cnt == N) {
        cnt -= 1;
        answer += 1;
      }
    } else {
      i += 1;
      cnt = 0;
    }
  }

  cout << answer;
  return 0;
}